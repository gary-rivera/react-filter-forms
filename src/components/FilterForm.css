.filter-form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: sticky;
  top: 0;
  background: #f8f8ff;
}

input:focus-visible {
  -webkit-appearance: none;
  outline: none;
  border-bottom: 2px solid rgba(128, 128, 128, 1);
}

input {
  /* appearance */
  -webkit-appearance: none;
  display: block;
  background: none;
  height: 3.5rem;
  width: 98%;
  margin: auto;
  border: none;
  border-bottom: 2px solid rgba(128, 128, 128, 0.25);

  /* font */
  font-family: 'Raleway', sans-serif;
  font-size: 1.5rem;
  font-weight: 300;
}
