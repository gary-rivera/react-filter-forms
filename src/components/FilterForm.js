import './FilterForm.css';
import { useState, useEffect } from 'react';

/** Form for filtering by name or tags of students.
 *
 * Props:
 * - filters (state from parent component to be updated by event handlers)
 * - addFilter (function to update main state for filters)
 *
 * { Main } -> FilterForm
 */

// TODO: position sticky to keep from scrolling within div?

function FilterForm({filters, addFilter}) {
  const [formData, setFormData] = useState(filters);
  
  function handleChange(evt) {
    const {name, value} = evt.target;
    setFormData(fData => ({
        ...fData,
        [name]: value
      }
    ))
  }

  useEffect(() => {
    addFilter(formData)
  }, [formData, filters, addFilter])
  
  return (
    <form className="filter-form">
      <input 
        label="name"
        name="name"
        htmlFor="name"
        value={formData.name} 
        placeholder="Search by name" 
        onChange={handleChange}
        autoComplete="off"
      />
      <input 
        label="tag"
        name="tag"
        htmlFor="tag"
        value={formData.tag} 
        placeholder="Search by tag" 
        onChange={handleChange}
        autoComplete="off"
      />
    </form>
  )
}

export default FilterForm;