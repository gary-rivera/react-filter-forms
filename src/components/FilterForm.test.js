import { render } from '@testing-library/react';
import FilterForm from './FilterForm';

const filtersTest = {
  "name": "",
  "tag": ""
}

function addFilterTest() {
  return
}

it('renders without crashing', function () {
  render(<FilterForm filters={filtersTest} addFilter={addFilterTest}/>);
});

it("renders the correct amount of input elements", () => {
  const { container } = render(<FilterForm 
                                filters={filtersTest} 
                                addFilter={addFilterTest}
                              />)
  const inputs = container.querySelectorAll('input')
  expect(inputs.length).toEqual(2);
});

it("matches snapshot", function () {
  const { container } = render(<FilterForm 
                                filters={filtersTest} 
                                addFilter={addFilterTest}
                              />)
  expect(container).toMatchSnapshot();
});