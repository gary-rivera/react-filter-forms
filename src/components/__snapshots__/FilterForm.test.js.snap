// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <form
    class="filter-form"
  >
    <input
      autocomplete="off"
      for="name"
      label="name"
      name="name"
      placeholder="Search by name"
      value=""
    />
    <input
      autocomplete="off"
      for="tag"
      label="tag"
      name="tag"
      placeholder="Search by tag"
      value=""
    />
  </form>
</div>
`;
